### PR TITLE
Bump logrus 1.4.2, go-shellwords, mergo, flock, creack/pty, golang/gddo, gorilla/mux

### LINKED ISSUE #25: Backtick "eats" all runes until isSpace
Hi @mattn , I found a thing while trying to use `go-shellwords`.

```
go test -v 
=== RUN   ExampleShellwords
--- FAIL: ExampleShellwords (0.02s)
got:
["-flag=val0" "val1" "-flag=val2"]
want:
["-flag=val0" "-flag=val1" "-flag=val2"]
FAIL
exit status 1
FAIL	testsw	0.041s
```

```
cat testsw_test.go 
package testsw

import (
	"fmt"
	"github.com/mattn/go-shellwords"
)

func ExampleShellwords() {
	parser := shellwords.NewParser()
	parser.ParseBacktick = true
	args, err := parser.Parse("-flag=val0 -flag=`cat testdata/flag-val.txt` -flag=val2")
	if err != nil {
		panic(err)
	}
	fmt.Printf("%q\n", args)
	// Output:
	// ["-flag=val0" "-flag=val1" "-flag=val2"]
}
```

```
cat go.mod 
module testsw

go 1.12

require github.com/mattn/go-shellwords v1.0.5
```

```
cat testdata/flag-val.txt 
val1
```

### LINKED ISSUE #27: Multi-commands inside of command substitution are throwing "invalid command line string" errors
Following commands throws errors: 
- `$(echo xxx | xargs)`
- `$(sleep 1 && sleep 1)`
- `$(sleep 1; sleep 1)`

Here is full runnable code: https://github.com/curusarn/go-shellwords-issue/blob/master/main.go

Output:
```
Commandline: $(sleep 1; sleep 1)
2019/08/10 22:19:46 invalid command line string
exit status 1
```

### LINKED ISSUE #33: incorrect interface
Flock does not exactly follow the sync.Locker interface, as it's Lock() method returns an error. sync.Mutex panics in certain occasions. I think we could have a version of flock which also panics instead of returning errors, to ensure we have the correct interface.

### LINKED ISSUE #111: WithOverride should be able to check types
Merging slices with the `WithOverride` option does not check that the slices have the same type. But before #96, one could use the `WithAppendSlice` option to have a "strict" override, failing if the types are different (check this test, failing now with mergo master https://github.com/docker/app/blob/master/types/parameters/merge_test.go#L10 ).

I propose to add an option, like `WithStrictOverride` to get this "feature" back, which will fail in this case.

### LINKED ISSUE #114: Embedded struct of pointer types will overwrite the whole destination struct
If you create a struct that contains multiple pointer types and then attempt to merge two of them together where only one of those types has a value, then the whole destination struct will be overwritten. 

Simple example
```go
type Config struct {
	Foo string `json:"foo"`
	Bar string `json:"bar"`
	Params *Params `json:"params"`

}
type Params struct{
	Name string `json:"name"`
	Multi *MultiPtr `json:"multi"`
	Final *Final `json:"Final"`
}
type MultiPtr struct {
	Text string `json:"text"`
	Number int `json:"number"`
}
type Final struct {
	Msg1 string `json:"msg"`
	Msg2 string `json:"msg_2"`
}
func main() {
	a := &Config{
		Foo: "something",
		Params: &Params{
			Name: "should remain untouched", // will be removed/destroyed if `mergo.WithOverride` is enabled
			Multi: &MultiPtr{ // will be removed/destroyed if `mergo.WithOverride` is enabled
				Text: "should stay since b MultiPtr is nil",
				Number: 5,
			},
			Final: &Final{
				Msg1: "will only be overwritten when `mergo.WithOverride` is enabled",
				Msg2: "", // will always overwritten
			},
		},
	}
	b := &Config{
		Bar: "from b",
		Params: &Params{
			Final: &Final{
				Msg1: "I overwrote something",
				Msg2: "thanks for looking into this",
			},
		},
	}
	// with mergo.WithOverride disabled
	//if err := mergo.Merge(a, b); err != nil {
	//	log.Println(err)
	//}
	// outputs: {"foo":"something","bar":"from b","params":{"name":"should remain untouched","multi":{"text":"should stay since b MultiPtr is nil"},"Final":{"msg":"will only be overwritten when `mergo.WithOverride` is enabled","msg_2":"thanks for looking into this"}}}

	// with mergo.WithOverride enabled
	if err := mergo.Merge(a, b, mergo.WithOverride); err != nil {
		log.Println(err)
	}
        // Outputs: {"foo":"something","bar":"from b","params":{"name":null,"multi":null,"Final":{"msg":"I overwrote something","msg_2":"thanks for looking into this"}}}

	j, _ := json.Marshal(a)
	fmt.Println(string(j))
}
```

I saw issue #47 and the PR that followed it, but that doesn't seem to solve this issue

### PR DESCRIPTION
**Pull request by thaJeztah**
### bump logrus v1.4.2

full diff: https://github.com/sirupsen/logrus/compare/v1.4.1...v1.4.2

- sirupsen/logrus#946 Fix solaris build
- sirupsen/logrus#966 Add a checkTerminal for nacl to support running on play.golang.org
- sirupsen/logrus#969 fix build break for plan9

### bump mattn/go-shellwords v1.0.6

full diff: https://github.com/mattn/go-shellwords/compare/v1.0.5...v1.0.6

relevant changes:

- mattn/go-shellwords#24 Add dir option for parser
- mattn/go-shellwords#26 Fix backquote in part of argument
    - fixes mattn/go-shellwords#25 Backtick "eats" all runes until isSpace
- mattn/go-shellwords#28 Fix dollar quote
    - fixes mattn/go-shellwords#27 Multi-commands inside of command substitution are throwing "invalid command line string" errors
- mattn/go-shellwords#24 Add dir option for parser
- mattn/go-shellwords#24 Add dir option for parser
- mattn/go-shellwords#24 Add dir option for parser

### bump github.com/imdario/mergo v0.3.8

full diff: https://github.com/imdario/mergo/compare/v0.3.7...v0.3.8

includes:

- imdario/mergo#112 Add strict override
    - fixes imdario/mergo#111 WithOverride should be able to check types
- imdario/mergo#106 Fix merging of interface types with concrete values
- imdario/mergo#120 should not overwrite pointers directly, instead check embedded values
    - fixes imdario/mergo#114 Embedded struct of pointer types will overwrite the whole destination struct
- imdario/mergo#125 added WithOverrideEmptySlice config flag

### bump gofrs/flock v0.7.1

full diff: https://github.com/gofrs/flock/compare/v0.7.0...v0.7.1

- gofrs/flock#34 don't mention sync.Locker in package documentation
    - fixes gofrs/flock#33 incorrect interface
- gofrs/flock#35 Fix linting issues and add goreportcard badge

### bump creack/pty v1.1.9

full diff: https://github.com/creack/pty/compare/v1.1.7...v1.1.9

adds go mod support

### bump olang/gddo 72a348e765d293ed6d1ded7b699591f14d6cd921

no local code changes, just keeping up with upstream

full diff: https://github.com/golang/gddo/compare/9b12a26f3fbd7397dee4e20939ddca719d840d2a...72a348e765d293ed6d1ded7b699591f14d6cd921

### bump gorilla/mux v1.7.3

full diff: full diff: https://github.com/gorilla/mux/compare/v1.7.2...v1.7.3

changes included:

- gorilla/mux#477 Improve CORS Method Middleware
    - implements gorilla/mux#477 Make CORSMethodMiddleware actually make sense
- gorilla/mux#489 Fix nil panic in authentication middleware example

### CODE REVIEW

File: vendor.conf [hunk 1]
```diff
@@ -33,7 +33,7 @@ github.com/opentracing/opentracing-go               1361b9cd60be79c4c3a7fa9841b3
 github.com/google/shlex                             6f45313302b9c56850fc17f99e40caebce98c716
 github.com/opentracing-contrib/go-stdlib            b1a47cfbdd7543e70e9ef3e73d0802ad306cc1cc
 github.com/mitchellh/hashstructure                  2bca23e0e452137f789efbc8610126fd8b94f73b
-github.com/gofrs/flock                              7f43ea2e6a643ad441fc12d0ecc0d3388b300c53 # v0.7.0
+github.com/gofrs/flock                              392e7fae8f1b0bdbd67dad7237d23f618feb6dbb # v0.7.1
```
Comment (thaJeztah) [inline vendor.conf]: Haven't opened a PR in buildkit for this yet; let me know if you want me to do that @tonistiigi

**Comment by thaJeztah** [comment]
@vdemeester ptal for the mergo bump

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM :lion:

**Pull request merged**